yaksh/views.py

```diff
@@ -759,30 +761,31 @@ def show_statistics(request, questionpaper_id, attempt_number=None):
 
 
 @login_required
-def monitor(request, questionpaper_id=None):
+def monitor(request, quiz_id=None):
     """Monitor the progress of the papers taken so far."""
 
     user = request.user
     ci = RequestContext(request)
     if not user.is_authenticated() or not is_moderator(user):
         raise Http404('You are not allowed to view this page!')
 
-    if questionpaper_id is None:
-        q_paper = QuestionPaper.objects.filter(Q(quiz__course__creator=user) |
-                                               Q(quiz__course__teachers=user),
-                                               quiz__is_trial=False
-                                               ).distinct()
-        context = {'papers': [],
-                   'quiz': None,
-                   'quizzes': q_paper}
+    if quiz_id is None:
+        course_details = Course.objects.filter(Q(creator=user) |
+                                           Q(teachers=user),
+                                           is_trial=False).distinct()
+        context = {'papers': [], "course_details": course_details,
+                   "msg": "Monitor"}
         return my_render_to_response('yaksh/monitor.html', context,
                                      context_instance=ci)
     # quiz_id is not None.
     try:
         q_paper = QuestionPaper.objects.filter(Q(quiz__course__creator=user) |
                                                Q(quiz__course__teachers=user),
                                                quiz__is_trial=False,
-                                               id=questionpaper_id).distinct()
+                                               quiz_id=quiz_id).distinct()
+        quiz = get_object_or_404(Quiz, id=quiz_id)
```

Comment:
Could this be in the beginning of the monitor view function? It seems to be oddly placed here.